ibis/backends/datafusion/__init__.py

```diff
@@ -105,7 +110,12 @@ def do_connect(
                 ).with_information_schema(True)
             else:
                 df_config = None
-            self.con = SessionContext(df_config)
+            if RuntimeConfig is None:
+                self.con = SessionContext(df_config)
+            else:
+                # datafusion 40.1.0 has a bug where SessionContext requires
+                # both SessionConfig and RuntimeConfig be provided.
+                self.con = SessionContext(df_config, RuntimeConfig())
```

Comment:
This is working around an upstream bug introduced in this release, see https://github.com/apache/datafusion-python/pull/827.